File: src/gestorAplicacion/economia/PrestamoFugaz.java
package gestorAplicacion.economia;

import gestorAplicacion.usuario.Usuario;

import java.time.LocalDate;

public class PrestamoFugaz extends Prestamo {

    private static final long serialVersionUID = -3388280595710146580L;

    public PrestamoFugaz(Usuario usuario, double valorInicial, LocalDate fechaInicio, Divisa divisa) {
        super(usuario,valorInicial, 6, fechaInicio, fechaInicio.plusMonths(6), divisa);
        setTEA(valorInicial);
    }

    @Override
    protected void setTEA(double monto) {
        if (monto < Divisa.COP.ConvertToDivisa(1000000, this.getDivisa())[0]) {
            super.TEA = super.baseTEAAlto;
        } else {
            super.TEA = super.baseTEABajo;
        }
    }

    @Override
    protected void setTEA(double monto, Garantia garantia) {
        this.setTEA(monto);
    }

    public double getTEA() {
        return TEA;
    }
}

File: Python/gestorAplicacion/economia/PrestamoFugaz.py
from gestorAplicacion.economia import Prestamo
from gestorAplicacion.economia import Divisa
from datetime import date, time, datetime

class PrestamoFugaz(Prestamo):

    _SERIALVERSIONUID = -3388280595710146580

    def __init__(self, usuario, valorInicial, fechaInicio, divisa):
        super().__init__(usuario,valorInicial, 6, fechaInicio, fechaInicio+datetime.timedelta(weeks=24), divisa)
        self.setTEA(valorInicial)

    def setTEA(self, monto):
        if monto < Divisa.COP.ConvertToDivisa(1000000, self.getDivisa())[0]:
            self.TEA = self.baseTEAAlto
        else:
            self.TEA = self.baseTEABajo

    def setTEA(self, monto, garantia):
        self.setTEA(monto)

    def getTEA(self):
        return self.TEA
